server.go

```diff
@@ -4173,13 +4327,297 @@ func (s *server) SubscribeCustomMessages() (*subscribe.Client, error) {
 	return s.customMessageServer.Subscribe()
 }
 
+// notifyOpenChannelPeerEvent updates the server's peer maps and then calls
+// the channelNotifier's NotifyOpenChannelEvent.
+func (s *server) notifyOpenChannelPeerEvent(op wire.OutPoint,
+	remotePub string) error {
+
+	// Call NewOpenChan to update the server's maps for this peer.
+	if err := s.NewOpenChan(remotePub); err != nil {
+		return err
+	}
+
+	// Notify subscribers about this open channel event.
+	s.channelNotifier.NotifyOpenChannelEvent(op)
+
+	return nil
+}
+
+// notifyPendingOpenChannelPeerEvent updates the server's maps and then calls
+// the channelNotifier's NotifyPendingOpenChannelEvent.
+func (s *server) notifyPendingOpenChannelPeerEvent(op wire.OutPoint,
+	pendingChan *channeldb.OpenChannel, remotePub string) error {
+
+	// Call NewPendingOpenChan to update the server's maps for this peer.
+	if err := s.NewPendingOpenChan(remotePub); err != nil {
+		return err
+	}
+
+	// Notify subscribers about this event.
+	s.channelNotifier.NotifyPendingOpenChannelEvent(op, pendingChan)
+
+	return nil
+}
+
+// notifyFundingTimeoutPeerEvent updates the server's maps and then calls the
+// channelNotifier's NotifyFundingTimeout.
+func (s *server) notifyFundingTimeoutPeerEvent(op wire.OutPoint,
+	remotePub string) error {
+
+	// Call NewPendingCloseChan to potentially demote the peer.
+	if err := s.NewPendingCloseChan(remotePub); err != nil {
+		return err
+	}
+
+	// Notify subscribers about this event.
+	s.channelNotifier.NotifyFundingTimeout(op)
+
+	return nil
+}
+
+// NewPendingOpenChan is called after the pending-open channel has been
+// committed to the database. This may transition a restricted-access peer to a
+// temporary-access peer.
+func (s *server) NewPendingOpenChan(remoteHex string) error {
```

Comment:
Ideally the ban logic could be separated into its own type altogether and so we could add extensive test coverage through unit tests.

Reply:
Can you elaborate? Do you mean like a new struct/pkg?

Reply:
If possible would be nice to decouple the peer banning logic from the `server` to its own type (struct) so we can cover it with tests separately.

Reply:
at least unify the two different ban-logics we currently have in place (proofOfChannel, banManager). Can we unit-test those functions?